lib/rubocop/cop/style/file_read.rb

```diff
@@ -0,0 +1,110 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Style
+      # Favor `File.(bin)read` convenience methods.
+      #
+      # @example text mode
+      #   # bad
+      #   File.open(filename).read
+      #   File.open(filename, &:read)
+      #   File.open(filename) { |f| f.read }
+      #   File.open(filename) do |f|
+      #     f.read
+      #   end
+      #   File.open(filename, 'r').read
+      #   File.open(filename, 'r', &:read)
+      #   File.open(filename, 'r') do |f|
+      #     f.read
+      #   end
+      #
+      #   # good
+      #   File.read(filename)
+      #
+      # @example binary mode
+      #   # bad
+      #   File.open(filename, 'rb').read
+      #   File.open(filename, 'rb', &:read)
+      #   File.open(filename, 'rb') do |f|
+      #     f.read
+      #   end
+      #
+      #   # good
+      #   File.binread(filename)
+      #
+      class FileRead < Base
+        extend AutoCorrector
+        include RangeHelp
+
+        MSG = 'Use `File.%<read_method>s`.'
+
+        RESTRICT_ON_SEND = %i[open].to_set.freeze
```

Comment:
Cosmetic `to_set` is redundant.

Reply:
I added it because of https://github.com/rubocop/rubocop-performance/issues/131

Reply:
I was unaware of that, but it turns out the set doesn't help much https://github.com/rubocop/rubocop/blob/de4f7cdd266999bf2cfbe78bd7fd8a0d4580863d/lib/rubocop/cop/base.rb#L305